makefile.shade

```diff
@@ -1,11 +1,65 @@
+use namespace="System.Diagnostics"
+use namespace="System.IO"
+
 var VERSION='0.1'
 var FULL_VERSION='0.1'
 var AUTHORS='Microsoft'
-var kestrelSrcDir='${Path.Combine(Directory.GetCurrentDirectory(), "src/Microsoft.AspNetCore.Server.Kestrel")}'
 
 use-standard-lifecycle
 k-standard-goals
 custom-goals
 
-#initialize if='Directory.Exists("src")'
-  exec program='dotnet' commandline='run -p ../../tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode Internal/Http/FrameHeaders.Generated.cs Internal/Http/Frame.Generated.cs' workingdir='${kestrelSrcDir}'
+#initialize
+  @{
+    if (Directory.Exists("src"))
+    {
+      Exec("dotnet",
+        commandline: "run -p ../../tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode Internal/Http/FrameHeaders.Generated.cs Internal/Http/Frame.Generated.cs",
+        workingdir: Path.Combine(Directory.GetCurrentDirectory(), "src/Microsoft.AspNetCore.Server.Kestrel"));
+    }
+
+    if (IsOSX())
+    {
+      var noParallelTestProjects = Environment.GetEnvironmentVariable("NO_PARALLEL_TEST_PROJECTS") ?? string.Empty;
+      noParallelTestProjects += ",Microsoft.AspNetCore.Server.Kestrel.FunctionalTests";
+      Environment.SetEnvironmentVariable("NO_PARALLEL_TEST_PROJECTS", noParallelTestProjects);
+    }
+  }
+
+functions @{
+  bool IsOSX()
+  {
+    try
+    {
+      var output = string.Empty;
+
+      var processStartInfo = new ProcessStartInfo
+      {
+        UseShellExecute = false,
+        FileName = "uname"
+      };
+
+      processStartInfo.RedirectStandardOutput = true;
+
+      using (var process = Process.Start(processStartInfo))
+      {
+        process.EnableRaisingEvents = true;
```

Comment:
Couldn't you `Output.ReadToEnd()` here instead of listening for events? `uname` doesn't have a lot of spew so this should be fine.